server/src/main/java/org/elasticsearch/ingest/IngestDocument.java

```diff
@@ -395,27 +495,109 @@ public void removeField(String path, boolean ignoreMissing) {
      * or can indicate a failure.
      */
     private static ResolveResult resolve(final String[] pathElements, final int limit, final String fullPath, Object context) {
+        return resolve(pathElements, limit, fullPath, context, IngestPipelineFieldAccessPattern.CLASSIC);
```

Comment:
It looks like this method is unused and can be removed (which would be good so that no one accidentally uses it in the future).